src/lib/saveUtils.ts

```diff
@@ -0,0 +1,73 @@
+import { getReportingPeriodDates } from './reportingPeriodDates'
+
+export function formatAsReportingPeriods(
+  entries: { year: number }[],
+  fiscalYear: { startMonth: number; endMonth: number },
+  category: 'emissions' | 'economy'
+) {
+  return entries.map(({ year, ...data }) => {
+    const [startDate, endDate] = getReportingPeriodDates(
+      year,
+      fiscalYear.startMonth,
+      fiscalYear.endMonth
+    )
+    return {
+      startDate,
+      endDate,
+      [category]:
+        category === 'economy'
+          ? (data as { economy: any }).economy
+          : {
+              ...data,
+            },
+    }
+  })
+}
+
+import { askPrompt } from './openai'
+
+export const defaultMetadata = (url: string) => ({
+  source: url,
+  comment: 'Parsed by Garbo AI',
+})
+
+export const askDiff = async (
+  existingCompany: any,
+  { scope12, scope3, biogenic, fiscalYear }: any
+) => {
+  if (!existingCompany.reportingPeriods?.length) return ''
+
+  const before = {
+    reportingPeriods: existingCompany.reportingPeriods.map(
+      ({ startDate, endDate, emissions }) => ({
+        startDate,
+        endDate,
+        emissions,
+      })
+    ),
+  }
+
+  const after = {
+    reportingPeriods: [
+      ...(scope12
+        ? formatAsReportingPeriods(scope12, fiscalYear, 'emissions')
+        : []),
+      ...(scope3
+        ? formatAsReportingPeriods(scope3, fiscalYear, 'emissions')
+        : []),
+      ...(biogenic
+        ? formatAsReportingPeriods(biogenic, fiscalYear, 'emissions')
+        : []),
```

Comment:
Since we have `scope12`, `scope3` or `biogenic` at once, this might be fine.

However, we also need to diff for `economy` data. Or does that happen in a separate function?